### DISCUSSION TITLE
MatchReset/RoundReset Should Have a persist variable option.

### DISCUSSION THREAD

**Discussion by PassiveBolt190** [ikemen-engine/Ikemen-GO#1390] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

Whenever you use the RoundReset(or MatchReset) it will never persist the variables even if you override the match states.

### Describe the solution you'd like

There should be an additional option that allows variables to persist.

### Additional context

For example: My character has a move that resets the round when he dies. However, the round reset also resets the variable so he can never die.

**Comment by PassiveBolt190** [comment]
If this is implemented the Var Persist option should not impact the state controller and cause it to change the RoundReset to MatchReset.

**Comment by NeatUnsou** [comment, marked as answer]
Implemented by #3279
